mllib/src/main/scala/org/apache/spark/mllib/evaluation/BinaryClassificationMetrics.scala

```diff
@@ -43,10 +43,13 @@ import org.apache.spark.sql.DataFrame
 @Since("1.0.0")
 class BinaryClassificationMetrics @Since("1.3.0") (
     @Since("1.3.0") val scoreAndLabels: RDD[(Double, Double)],
-    @Since("1.3.0") val numBins: Int) extends Logging {
+    @Since("1.3.0") var numBins: Int) extends Logging {
```

Comment:
hi @srowen , can we use a simple way to add the setBins method.  Seems add a setBins method is useful. 
Just as the testSuite in this PR shows. If the user wants to test different numBins. 
Create one object, just call setBins, he can easily get different results. 
How about change a way to implement setBins?